src/libraries/Microsoft.PowerFx.Core/Types/DType.cs

```diff
@@ -2709,6 +2725,12 @@ public static DType Union(ref bool fError, DType leftType, DType rightType, bool
             leftType.AssertValid();
             rightType.AssertValid();
 
+            if ((leftType.IsSealed || rightType.IsSealed) && leftType != rightType)
```

Comment:
If we are union "to left type", would it be ok if the right type is sealed but the left type is not?